include/ttmlir/Dialect/TT/IR/TTOpsTypes.td

```diff
@@ -51,6 +51,10 @@ def TT_ArchAttr : EnumAttr<TT_Dialect, TT_Arch, "arch"> {
   let assemblyFormat = "`<` $value `>`";
 }
 
+def TT_DataTypeAttr : EnumAttr<TT_Dialect, TT_DataType, "supportedDataTypes"> {
+  let assemblyFormat = "`<` $value `>`";
+}
+
```

Comment:
I think we should leave the vanilla enum attr that gets autogenerated, i.e. remove `let genSpecializedAttr = 0;` above.

Reply:
We found this was needed to prevent compile errors (copied what was done from TT_Arch in this same file, which had genSpecializedAttr = 0)

Reply:
I see, so at the very least we need to rename `"supportedDataTypes"` to `"data_type"`, perhaps we can drop the brackets `<` and `>`? Not sure if tablegen will like that or not